### PR TITLE
Installer: Check for an existing installation, skip applying

### PR DESCRIPTION
**Pull request by ark3**
Please comment/advise on FIXME and TODO comments...

The diff is easiest to read with whitespace changes ignored.

Also: This is only very lightly tested.

### CODE REVIEW

File: cmd/edgectl/aes_install.go [hunk 1]
```diff
@@ -319,6 +319,50 @@ func (i *Installer) CheckACMEIsDone(hostname string) error {
...
 }
 
+// GetAESCRDs returns the names of the AES CRDs available in the cluster (and
+// logs them as a side effect)
+func (i *Installer) GetAESCRDs() ([]string, error) {
+	crds, err := i.CaptureKubectl("get AES crds", "", "get", "crds", "-lproduct=aes", "-o", "name")
+	if err != nil {
+		return nil, err
+	}
+	res := make([]string, 0, 15)
```
Comment (alexgervais) [inline cmd/edgectl/aes_install.go]: Is there a reason to use 15 cap? It seems brittle and could cause issues for misconfigured/duplicated CRDs and won't play nicely if we ever reach that number of CRDs.

File: cmd/edgectl/aes_install.go [hunk 2]
```diff
@@ -809,6 +901,19 @@ Timed out waiting for the load balancer's IP address for the AES Service.
 
 const tryAgain = "If this appears to be a transient failure, please try running the installer again. It is safe to run the installer repeatedly on a cluster."
 
+const abortExisting = `
+This tool does not support upgrades/downgrades at this time.
+
+Aborting the installer to avoid corrupting an existing installation of AES.
+`
+
+// TODO: Explain how to remove existing CRDs? Or do something smarter?
```
Comment (alexgervais) [inline cmd/edgectl/aes_install.go]: Yes, I think we should have instructions for removing the existing CRDs and resuming using the installer 👍

File: cmd/edgectl/aes_install.go (hunk 1 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: Have I misunderstood how that works? I thought I was just suggesting an initial capacity for the slice, but that `append` will increase it (by allocating a new array) as needed.

File: cmd/edgectl/aes_install.go (hunk 2 above)
Comment (ark3) [reply cmd/edgectl/aes_install.go]: I couldn't come up with those instructions quickly, so I punted to a TODO. Thoughts? 

- `kubectl delete ambassador-crds` doesn't work
- `kubectl delete crd -l product=aes` only works if the CRDs are installed using our YAML. 
- `kubectl delete $(kubectl get crd -o name | grep "[.]getambassador[.]io$")` doesn't work on Windows

File: cmd/edgectl/aes_install.go (hunk 1 above)
Comment (alexgervais) [reply cmd/edgectl/aes_install.go]: You are right, append will reallocate the array.

File: cmd/edgectl/aes_install.go (hunk 2 above)
Comment (alexgervais) [reply cmd/edgectl/aes_install.go]: I would do `kubectl delete crd -l product=aes`. Assuming the listed CRDs are also visible in the terminal output, _most_ users should be able to figure it out.

Comment (ark3) [reply cmd/edgectl/aes_install.go]: Oh, the other issue is that deleting the CRDs also deletes and CRs of those types. Need to warn about that!

**Comment by alexgervais** [comment]
Validated:
✔️ Empty cluster install
✔️ Completing the install after previously having installed the correct AES version
✔️ Handling "upgrade" path
```
edgectl install
-> Installing the Ambassador Edge Stack 1.3.0.
-> Found an existing installation of Ambassador Edge Stack 1.2.2.

This tool does not support upgrades/downgrades at this time.

Aborting the installer to avoid corrupting an existing installation of AES.

See https://www.getambassador.io/user-guide/getting-started/

Full logs at /var/folders/n6/w_qkrxw148j_l7h_45z35c0r0000gn/T/edgectl-install-20200323-150118.log

Error: install: existing AES 1.2.2 found when installing AES 1.3.0
```
✔️ Handling installing just the CRDs
```
edgectl install     
-> Installing the Ambassador Edge Stack 1.3.0.
-> Found Ambassador CRDs in your cluster, but no AES installation.

You can manually remove installed CRDs if you are confident they are not in use
by any installation.
Removing the CRDs will cause your existing Ambassador Mappings and other
resources to be deleted as well.

$ kubectl delete crd -l product=aes

Aborting the installer to avoid corrupting an existing (but undetected)
installation.

See https://www.getambassador.io/user-guide/getting-started/

Full logs at /var/folders/n6/w_qkrxw148j_l7h_45z35c0r0000gn/T/edgectl-install-20200323-150558.log

Error: install: CRDs found
```

**Pull request merged**